third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -170,6 +182,11 @@ impl Interpreter {
                         .map_err(|e| self.set_location(e))?;
 
                     if let Some(frame) = self.call_stack.pop() {
+                        if frame.function.module_id() != current_frame.function.module_id() {
```

Comment:
why?
And you can chain then I think
https://rust-lang.github.io/rfcs/2497-if-let-chains.html

Reply:
And this applies to all the function call? Would this degrade perf?

Reply:
This applies to all function call indeed. See the AIP thread for detailed reason why we need this check. We can measure the performance degredation for sure but I think it shouldn't be too bad?

Reply:
This is not stablized yet?